core/vm/contracts.go

```diff
@@ -130,6 +131,10 @@ var PrecompiledContractsBernoulli = map[common.Address]PrecompiledContract{
 // PrecompiledContractsDescartes returns the default set of precompiled contracts,
 // including the L1SLoad precompile.
 func PrecompiledContractsDescartes(cfg Config) map[common.Address]PrecompiledContract {
+	if cfg.L1Client == nil || (reflect.ValueOf(cfg.L1Client).Kind() == reflect.Ptr && reflect.ValueOf(cfg.L1Client).IsNil()) {
+		// TODO: have a better way to handle this error
+		panic("No L1 client provided in the vm config")
+	}
```

Comment:
just a nil check will do, the reflect business is really unnecessary.

Reply:
fixed